### PR TITLE
[Connectors API] Make nullable fields optional in the parser inside ConnectorSyncJob

### PR DESCRIPTION
**Pull request by timgrein**
Adjust some fields in the connector parser inside `ConnectorSyncJob` to reflect that they're nullable.

### CODE REVIEW

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/Connector.java [hunk 1]
```diff
@@ -153,7 +153,7 @@ private Connector(
         this.error = error;
         this.features = features;
         this.filtering = Objects.requireNonNull(filtering, "[filtering] cannot be null");
-        this.indexName = Objects.requireNonNull(indexName, "[index_name] cannot be null");
```
Comment (timgrein) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/Connector.java:156]: As `indexName` is annotated with `@Nullable` I think we should not have this check. (led to a failure in one of the new test cases, so I simply fixed it directly with this PR).

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJob.java [hunk 2]
```diff
@@ -359,6 +359,14 @@ public static ConnectorSyncJob fromXContent(XContentParser parser) throws IOExce
         return PARSER.parse(parser, null);
     }
 
+    public static Connector syncJobConnectorFromXContentBytes(BytesReference source, XContentType xContentType) {
```
Comment (timgrein) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJob.java:362]: Using this method inside `ConnectorSyncJobIndexService` will be a separate PR as this needs some more thinking how to correctly handle the `id` of the connector (when, where and how to set it).

Comment (jedrazb) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJob.java:381]: Sure, if you want to pass connector_id to the connector parser from context you can you can change ` SYNC_JOB_CONNECTOR_PARSER.parse(parser, null);` to `SYNC_JOB_CONNECTOR_PARSER.parse(parser, docId);` and set the Class `id` param in `ConstructingObjectParser`

**Review by jedrazb — APPROVED** [review]

**Review by jedrazb — APPROVED** [review]

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJob.java (hunk 2 above)
Comment (timgrein) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJob.java:381]: I actually went down this path already and encountered one minor issue/inconvenience, but nothing blocking. We can discuss that in a 1on1 👍

**Comment by jedrazb** [comment]
@timgrein that's a bug that affects 8.12 as well, we need to backport it. I'm adding a `>bug` label and create a manual backport

**Comment by jedrazb** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.12|[](https://github.com/elastic/elasticsearch/pull/103544)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)
